esphome/components/adc/adc_sensor_esp32_v5.cpp

```diff
@@ -0,0 +1,393 @@
+#ifdef USE_ESP32
+
+#include "adc_sensor.h"
+#include "esphome/core/log.h"
+
+#if (ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0))
+
+namespace esphome {
+namespace adc {
+
+static const char *const TAG = "adc.esp32";
+
+adc_oneshot_unit_handle_t ADCSensor::shared_adc1_handle = nullptr;
+adc_oneshot_unit_handle_t ADCSensor::shared_adc2_handle = nullptr;
+
+void ADCSensor::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up ADC '%s'...", this->get_name().c_str());
+
+  if (!this->do_setup_) {
+    return;
+  }
+
+  this->init_complete_ = false;
+  this->handle_init_complete_ = false;
+  this->config_complete_ = false;
+  this->calibration_complete_ = false;
+
+  if (this->is_adc1_) {
+    if (this->adc1_handle_ == nullptr) {
+      // Check if another sensor already initialized ADC1
+      if (shared_adc1_handle != nullptr) {
+        this->adc1_handle_ = shared_adc1_handle;
```

Comment:
Maybe prefix these so we know they are static without guessing when looking at the code?
```suggestion
        this->adc1_handle_ = ADCSensor::shared_adc1_handle;
```